modules/caption-impl/src/main/java/org/opencastproject/caption/converters/WebVttCaptionConverter.java

```diff
@@ -107,7 +122,28 @@ public String getExtension() {
 
   @Override
   public Type getElementType() {
-    return MediaPackageElement.Type.Attachment;
+    return mediapackageElementType;
+  }
+
+  /**
+   * Converts the configured property 'mediapackage-element-type' into the corresponding enum 'MediapackageElement.Type'
+   *
+   * @param cc ComponentContext
+   * @return the configured mediapackage element type
+   */
+  private MediaPackageElement.Type getConfiguredMediapackageElementType(ComponentContext cc) {
+    String mediapackageElementType = null;
+    try {
+      mediapackageElementType = OsgiUtil.getOptCfg(cc.getProperties(), MEDIAPACKAGE_ELEMENT_TYPE_CONFIG_KEY).get();
+      // capitalize the first letter
+      mediapackageElementType = mediapackageElementType.substring(0, 1).toUpperCase()
+          + mediapackageElementType.substring(1);
```

Comment:
If you want to ignore casing, you need to convert this as well:

```suggestion
          + mediapackageElementType.substring(1).toLowerCase();
```

Reply:
For this, I switched to org.apache.commons.lang3.EnumUtils.